enhancements/cluster-scope-secret-volumes/shared-resources-via-openshift-builds-operator.md

```diff
@@ -0,0 +1,357 @@
+  `Enabled`, the operator does not install Shipwright Builds and reports that OpenShift Pipelines
+  should be installed in its status.
+- If Shipwright Builds is enabled and then later disabled, the following should occur:
+  - Shipwright build controllers should be deleted.
+  - CRDs and associated admission webhooks should remain in place.
+
+
+#### Shared Resource CSI Driver
+
+- To disable the Shared Resource CSI Driver by default, set the `SHARED_RESOURCES_DEFAULT_DISABLED`
+  environment variable to `true` in the operator `Subscription` object:
+
+  ```yaml
+  kind: Subscription
+  apiVersion: operators.coreos.com/v1alpha1
+  metadata:
+    name: openshift-build
+    namespace: openshift-operators
+  spec:
+    ...
+    config:
+      env:
+        - name: SHARED_RESOURCES_DEFAULT_DISABLED
+          value: "true"
+  ```
+
+- If the CSI driver is disabled after it has been enabled, the following should occur:
+  - The `CSIDriver` object for the driver should be deleted.
+  - The `DaemonSet` for the CSI driver should be deleted.
+  - CRDs and associated admission webhooks should remain in place.
+
+
+### API Extensions
+
+The OpenShift Builds will introduce a new configuration CRD, `OpenShiftBuild`. It will contain 
+`spec` fields that configure the Shared Resource CSI driver in addition to Shipwright-related
+components. The operator will also deploy (directly or indirectly) the following:
+
+- Custom resources for the Shipwright Operator
+- Custom resources and webhooks for Shipwright Build
+- Custom resources and webhooks for the Shared Resource CSI Driver
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Update domain name for Shared Resource CSI Driver
+
+The root domain/api group used for the Shared Resource CSI driver components will be changed from
+`sharedresource.openshift.io` to `sharedresource.dev`, impacting the registered name of the CSI
+driver and resource CRDs. To keep the existing feature in OCP, the Shared Resource CSI driver code
+may need to be forked to a new GitHub repo/organization.
+
+
+#### Dependency Resolution with OpenShift Pipelines
+
+Notably, the design of this operator does **_not_** intend to use OLM API dependency resolution
+to automate the deployment and management of OpenShift Pipelines. Cluster admins will need to
+separately (and manually) install OpenShift Pipelines. This ensures that OpenShift Pipelines is not
+installed if a cluster admin does not want Shipwright Builds.
+
+
+### Risks and Mitigations
+
+#### Duplicate CSI Drivers
+
+On a tech preview cluster, the same CSI driver could be deployed twice on tech preview clusters.
+Changing the root domain of the driver and CRDs avoids this problem - the driver and CRDs have
+different names/APIGroups, and this are distinct entities in OpenShift. The driver DaemonSet should
+likewise not collide, since it will be deployed in a different namespace.
```

Comment:
@gabemontero want to check here that the CSI driver doesn't use any hard-coded paths on the node that would be shared if we had two of these drivers running simultaneously, under different DaemonSets.

Reply:
good catch @adambkaplan .... there are some hard coded paths on the node with the current OCP driver, so we'll have to make some set of changes with the "operator delivered GA version" ... some combination of
- make different with the new version
- make configurable
- or the new version detects the presence of the old OCP version somehow and adjusts

Reply:
To install the old CSI driver, one needs `TechPreviewNoUpgrade` feature set, which implies no upgrade and the cluster is sort of testing-only. So, do we need to worry about anyone having both CSI drivers?

> The driver DaemonSet should likewise not collide, since it will be deployed in a different namespace.
The CSI driver name (i.e. `csi.sharedresource.openshift.io`) should be different too, just to be sure.

Reply:
There are two rationales for the rename:

1. Tactical: this gives us a wider install base. We've been promising this thing for a long time, so the more versions we support, the better. I agree that having both CSI drivers is a bit of an edge case.
2. Strategic: we will want to upstream this, either as its own project outright or as something within Kubernetes SIG-Storage. Getting rid of openshift-isms helps with that.